kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/ingress/ProxyIngressModel.java

```diff
@@ -40,21 +40,31 @@ public Optional<VirtualClusterIngressModel> clusterIngressModel(VirtualKafkaClus
                 .findFirst();
     }
 
-    public record VirtualClusterIngressModel(VirtualKafkaCluster cluster, List<IngressModel> ingressModels) {
+    /**
+     *
+     * @param cluster cluster
+     * @param ingressModelResults the ingress model results, one per ingress in the VKC spec.ingresses in the same order
+     */
+    public record VirtualClusterIngressModel(VirtualKafkaCluster cluster, List<IngressModelResult> ingressModelResults) {
 
         public Stream<Service> services() {
-            return ingressModels.stream().flatMap(it -> it.ingressInstance().services()).map(ServiceBuilder::build);
+            return ingressModelResults.stream().flatMap(it -> it.ingressModel().services()).map(ServiceBuilder::build);
         }
 
         public Set<IngressConflictException> ingressExceptions() {
-            return ingressModels.stream().flatMap(it -> Stream.ofNullable(it.exception)).collect(Collectors.toSet());
+            return ingressModelResults.stream().flatMap(it -> Stream.ofNullable(it.exception)).collect(Collectors.toSet());
```

Comment:
This looked rather odd initially, but having poked it in an IDE the alternative is:
```suggestion
            return ingressModelResults.stream().filter(it -> it.exception != null).map(IngressModelResult::exception).collect(Collectors.toSet());
```

which is a more verbose it might be a tiny bit more readable.